rpcserver.go

```diff
@@ -585,7 +586,7 @@ func (r *rpcServer) ListBatches(_ context.Context,
 		}
 
 	case len(req.GetBatchKeyStr()) > 0:
-		batchKeyBytes, err := hex.DecodeString(req.GetBatchKeyStr())
+		batchKeyBytes, err = hex.DecodeString(req.GetBatchKeyStr())
```

Comment:
IMO it's fine to shadow the `err` variable here, as we check it within the same scope. It would be different if we had a shared `if err != nil` after the `switch`. So this moves the `batchKeyBytes` outside of this scope which IMO just makes it a bit harder to understand (since it's not used anywhere else).

Reply:
Yeah, I guess shadowing is OK given that we always check errs right after assignment.